### PR TITLE
Enable typechecks for torch.nn.modules.[activation|upsampling]

### LINKED ISSUE #42986: Enable torch.nn.modules typechecks during CI
`TestTypeHints.test_run_mypy_strict` should pass with `[mypy-torch.nn.modules.*]` deleted from `mypy.ini`

cc @ezyang @malfet @rgommers

### PR DESCRIPTION
**Pull request by malfet**
Add missing `hardsigmoid`, `silu`, `hardswish` and `multi_head_attention_forward` to functional.pyi.in
 Embed some typing annotations into functional.py

### CODE REVIEW

File: torch/nn/functional.py
Comment (rgommers) [inline torch/nn/functional.py:3879]: You need an import for this, CI is failing with `NameError: name 'Tuple' is not defined`

**Review by rgommers — COMMENTED** [review]
LGTM, once CI is happy it should be good to merge.

File: torch/nn/functional.py
Comment (malfet) [reply torch/nn/functional.py:3879]: Oops, sorry update that. 
By the way, do you know why many files in .nn import typing from `_jit_internal.py` rather than from typing directly?

**Review by ezyang — APPROVED** [review]

File: torch/nn/functional.pyi.in
Comment (walterddr) [inline torch/nn/functional.pyi.in:171]: if we are changing them here, should we also add typing on sigmoid, hardsigmoid, silu in functional.py too?

Comment (malfet) [reply torch/nn/functional.pyi.in:171]: @walterddr those functions are already annotated (and exist in functional.py)

**Pull request closed without merging**
